### PR TITLE
Final refactor - rename mergerx to tuple-merger

### PR DESCRIPTION
**Pull request by tsafin**
Preparing to publication we rename module to be `tuple-merger` and not the ugly `mergerx.merger`.

* Renamed all former references to mergerx_ anything;
* Also used @Totktonada approach to embed Lua script which is necessary for post-load metatable initialization;

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
 if (APPLE)
-    set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+    set_target_properties(${LIBNAME} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
 endif()
 
 # Install module
-install(FILES mergerx.lua DESTINATION ${TARANTOOL_INSTALL_LUADIR}/${PROJECT_PREFIX}/)
-install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${TARANTOOL_INSTALL_LIBDIR}/${PROJECT_PREFIX})
+# install(FILES postload.lua DESTINATION ${TARANTOOL_INSTALL_LUADIR}/${LIBSUBDIR}/)
+install(TARGETS ${LIBNAME} LIBRARY DESTINATION ${TARANTOOL_INSTALL_LIBDIR}/${LIBSUBDIR})
```
Comment (Totktonada) [inline src/CMakeLists.txt:30]: 1. postload.lua is bundled, so we don't need to install it separately. We can remove the line.
2. AFAIS, LIBNAME and LIBSUBDIR are not defined.

File: tuple-merger-scm-1.rockspec [hunk 2]
```diff
@@ -1,20 +1,20 @@
-package = 'mergerx'
+package = 'tuple-merger'
 version = 'scm-1'
 
 -- url and branch of the package's repository at GitHub
 source  = {
-    url    = 'git://github.com/tsafin/tarantool-merge.git';
+    url    = 'git://github.com/tarantool/tuple-merger.git';
     branch = 'tarantool-merge';
```
Comment (Totktonada) [inline tuple-merger-scm-1.rockspec:7]: Should be 'master' in the foreseeable future?

**Review by Totktonada — APPROVED** [review]
Looked briefly — no objections (just several remarks, but CI likely catched them already).

File: src/CMakeLists.txt (hunk 1 above)
Comment (tsafin) [reply src/CMakeLists.txt:30]: 1. Yup, will remove
2. Thy were defined at the top

https://github.com/tsafin/tarantool-merge/blob/6d901c9b87d0dd5a2eca411c67977eae6c32ac47/src/CMakeLists.txt#L1-L2

File: tuple-merger-scm-1.rockspec (hunk 2 above)
Comment (tsafin) [reply tuple-merger-scm-1.rockspec:7]: Thanks for the catch! Yes, indeed!

**Pull request merged**